monitor/crawler.py

```diff
@@ -151,15 +157,15 @@ def clear(self, metadata: bool = True, certificates: bool = True) -> None:
 
         super().clear(metadata=metadata, certificates=certificates)
 
-    def store_state_metadata(self, state: dict):
+    def store_state_metadata(self, state: ArchivedFleetState):
         # TODO Limit the size of this table - no reason to store really old state values
 
-        db_row = (state['nickname'],
-                  state['symbol'],
-                  state['color_hex'],
-                  state['color_name'],
+        db_row = (str(state.nickname),
+                  state.nickname.characters[0].symbol,
+                  state.nickname.characters[0].color_hex,
+                  state.nickname.characters[0].color_name,
```

Comment:
I'm starting to think it might be better to return an actual documentable object from `status_info()`, with a `to_json()` method. Using JSON for interfacing inside Python seems like a bad idea.

> Looks like characters length is 2

Nope, state nickname is only 1 character. I have to do this since I don't want to change the DB schema, but in general it would be better to make it length-agnostic.

Reply:
Sounds like a job for a named tuple.

Reply:
We do need it to be json-serializable, but we can just overload the named tuple and add a method. Perhaps a dataclass would fit better, but we have to support Py3.6

Reply:
Yes, you're right. I don't forsee any issues supporting 3.6+ only.

Reply:
I meant, dataclasses require 3.7+ and we support 3.6+. But making a derived class from a namedtuple shouldn't be a problem.

Reply:
Ah, right. Well same goes for 3.7+.  In fact CI is already using that version.  No strong preference though, it's the namespacing that matters most I think.

Reply:
> Nope, state nickname is only 1 character. I have to do this since I don't want to change the DB schema, but in general it would be better to make it length-agnostic.

Whoops read the wrong Nickname usage. Yes, my hope was to figure out a way to make it length agnostic

Reply:
Actually I now realize that the DB here is transient, so we can change the schema. Maybe we should.